src/settings/eeprom/rf_channel.rs

```diff
@@ -113,40 +144,50 @@ impl Default for ChannelSettings {
             //
             // All of the power meters are preceded by attenuators which are incorporated in
             // the offset.
-            output_power_transform: LinearTransformation::new(1.0 / 0.035, -35.6 + 19.8 + 10.0),
+            output_power_transform: Leaf::from(LinearTransformation::new(
+                1.0 / 0.035,
+                -35.6 + 19.8 + 10.0,
+            )),
 
             // The input power and reflected power detectors have an op-amp gain of 1.5
-            reflected_power_transform: LinearTransformation::new(
+            reflected_power_transform: Leaf::from(LinearTransformation::new(
                 1.0 / 1.5 / 0.035,
                 -35.6 + 19.8 + 10.0,
-            ),
+            )),
 
-            input_power_transform: LinearTransformation::new(1.0 / 1.5 / 0.035, -35.6 + 8.9),
+            input_power_transform: Leaf::from(LinearTransformation::new(
+                1.0 / 1.5 / 0.035,
+                -35.6 + 8.9,
+            )),
         }
     }
 }
 
 impl ChannelSettings {
-    fn validate_bias_voltage(&mut self, new: f32) -> Result<f32, &'static str> {
-        if (0.0..=platform::BIAS_DAC_VCC).contains(&(-1.0 * new)) {
-            Ok(new)
+    fn validate_bias_voltage(&mut self, depth: usize) -> Result<usize, &'static str> {
```

Comment:
This is problematic since the value has still changed.
Better clamp it silently.

Reply:
> Better clamp it silently.

How silent exactly?

Always clamp it and return `Ok`, or only clamp it when out of range and still return `Err`?